Firestore/core/test/firebase/firestore/nanopb/pretty_printing_test.cc

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright 2019 Google
+ *
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/Protos/nanopb/firestore/local/maybe_document.nanopb.h"
+#include "Firestore/Protos/nanopb/firestore/local/mutation.nanopb.h"
+#include "Firestore/Protos/nanopb/firestore/local/target.nanopb.h"
+#include "Firestore/Protos/nanopb/google/firestore/v1/document.nanopb.h"
+#include "Firestore/Protos/nanopb/google/firestore/v1/firestore.nanopb.h"
+#include "Firestore/Protos/nanopb/google/firestore/v1/write.nanopb.h"
+#include "Firestore/core/src/firebase/firestore/nanopb/message.h"
+#include "Firestore/core/src/firebase/firestore/nanopb/nanopb_util.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace firebase {
+namespace firestore {
+namespace nanopb {
+namespace {
+
+using ::testing::MatchesRegex;
+
+TEST(PrettyPrintingTest, PrintsInt) {
+  Message<firestore_client_WriteBatch> m;
+  m->batch_id = 123;
+
+  EXPECT_THAT(m.ToString(), MatchesRegex(
+                                R"(<WriteBatch 0x[0-9A-Fa-f]+>: {
+  batch_id: 123
+})"));
+}
+
+TEST(PrettyPrintingTest, PrintsBool) {
+  Message<firestore_client_MaybeDocument> m;
+  m->has_committed_mutations = true;
+
+  EXPECT_THAT(m.ToString(), MatchesRegex(
```

Comment:
You could slightly cut the verbosity down by writing a matcher.

Something like this:
```c++
MATCHER_P(HasToString, expected, "") {
  std::string arg_string = arg.ToString();
  return testing::Value(arg_string, MatchesRegex(expected));
}
```

This would allow you to write this as:

```c++
EXPECT_THAT(m, HasToString(R"(<MaybeDocument ..."));
```

It's not a big win, but would reduce some of the incantation burden.